modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -465,16 +469,22 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
       std::string plane_id_name = "plane_id";
       params.set<std::string>("id_name") = "plane_id";
 
-      _build_mesh = &addMeshSubgenerator("PlaneIDMeshGenerator", name() + "_extrudedIDs", params);
+      build_mesh_name = name() + "_extrudedIDs";
+      addMeshSubgenerator("PlaneIDMeshGenerator", build_mesh_name, params);
     }
   }
   else
     declareMeshProperty("extruded", false);
+
+  _build_mesh = &getMeshByName(build_mesh_name);
 }
 
 std::unique_ptr<MeshBase>
 PinMeshGenerator::generate()
 {
+  // Must be called to free the ReactorMeshParams mesh
+  freeReactorMeshParams();
```

Comment:
let s work to remove those. It s just odd to have to do this in C++

Reply:
I disagree. The problem is the oddity of the parameter MG that doesn't do anything.

Reply:
It's a database of properties. And it fits snuggly in the Mesh block, doesnt shock me

Reply:
Yeah. I can't agree that it's not clean from a user prospective. Just difficult from a developer prospective. Since it doesn't even provide a mesh